### PR TITLE
Rename to ClusterType and restore devnet compat.

### LINKED ISSUE #12039: Refactor OperatingMode
The `OperatingMode` type is a source of confusion for devs as it requires mentally mapping from the names it uses to the real clusters it represents.   This would be much more obvious way to express the different configuration options:

```diff
diff --git a/sdk/src/genesis_config.rs b/sdk/src/genesis_config.rs
index 4caa8d021..9383ebb18 100644
--- a/sdk/src/genesis_config.rs
+++ b/sdk/src/genesis_config.rs
@@ -32,9 +32,10 @@ use std::{
 pub const UNUSED_DEFAULT: u64 = 1024;
 
 #[derive(Serialize, Deserialize, Debug, Clone, Copy, PartialEq, AbiEnumVisitor, AbiExample)]
pub enum OperatingMode {
-    Preview,     // Next set of cluster features to be promoted to Stable
-    Stable,      // Stable cluster features
-    Development, // All features (including experimental features)
+    Testnet,      // Activate testnet.solana.com configuration
+    MainnetBeta,  // Activate mainnet-beta.solana.com configuration
+    Devnet,       // Activate devnet.solana.com configuration
+    AllFeatures   // A configuration suitable for local development, all features enabled
 }
 
 #[frozen_abi(digest = "2KQs7m2DbLxkEx6pY9Z6qwYJAhN2Q4AdoNgUcULmscgB")]
```

### PR DESCRIPTION
**Pull request by ryoqun**
#### Problem

OperatingMode doesn't match with the reality.

#### Summary of Changes

Refactor and restore compatibility for devnet.

I'll create parallel prs against ~v1.2~ and [v1.3](#12069 ) to apply them at once. (Edit: I figured out backporting this should wait until @jackcmay's master backporting work (#11568) is completed first to avoid needless large merge conflicts due to inverted cherry-pick order)  

Sibling: #12069

Fixes #12039

### CODE REVIEW

**Comment by ryoqun** [comment]
Wow, the test passed without any additional commits.. I can't believe it... ;)

File: genesis-programs/src/lib.rs [hunk 1]
```diff
@@ -84,16 +88,26 @@ fn get_programs(operating_mode: OperatingMode) -> Vec<(Program, Epoch)> {
         .into_iter()
         .map(|program| (program, GENESIS_EPOCH))
         .collect::<Vec<_>>(),
+        ClusterType::Devnet => vec![
+            // Programs used for testing
+            Program::BuiltinLoader(solana_bpf_loader_deprecated_program!()),
```
Comment (ryoqun) [inline genesis-programs/src/lib.rs:93]: it seems that `solana_bpf_loader_program!()` is yet to be added on devnet/v1.2

File: net/net.sh [hunk 2]
```diff
@@ -779,12 +779,12 @@ while [[ -n $1 ]]; do
     elif [[ $1 = --faucet-lamports ]]; then
       genesisOptions="$genesisOptions $1 $2"
       shift 2
-    elif [[ $1 = --operating-mode ]]; then
+    elif [[ $1 = --cluster-type ]]; then
```
Comment (ryoqun) [inline net/net.sh:782]: I went aggressive here not to retain compatibility, assuming these scripts isn't used for our any live clusters.

**Comment by ryoqun** [comment]
@mvines Could you review this and #12069?

> I'll create parallel prs against ~v1.2~ and [v1.3](#12069) to apply them at once. (Edit: I figured out backporting this should wait until @jackcmay's master backporting work (#11568) is completed first to avoid needless large merge conflicts due to inverted cherry-pick order)

Also, please note this.

I think it's acceptable for both `ClusterType` and `OperatingMode` to co-exist between maintained branches for small time window (until #11568) because of this. That's because generally renaming newly-backported PRs with a gating should be easier than addressing the avoidable large merge conflicts.

**Review by mvines — APPROVED** [review]
This is much nicer.

I'd like to upgrade devnet to 1.3 tomorrow ideally, so let's get this and https://github.com/solana-labs/solana/pull/12069 landed please!

File: genesis-programs/src/lib.rs (hunk 1 above)
Comment (jackcmay) [reply genesis-programs/src/lib.rs:93]: That's right, now backported yet

**Pull request merged**